src/commands/local-store/local-store-command.js

```diff
@@ -75,6 +81,110 @@ class LocalStoreCommand extends Command {
+                            );
+                            break;
+                        } catch (error) {
+                            attempts += 1;
+                            if (attempts < LOCAL_INSERT_MAX_ATTEMPTS) {
+                                // eslint-disable-next-line no-await-in-loop
+                                await setTimeout(LOCAL_INSERT_RETRY_DELAY);
+                            } else {
+                                // eslint-disable-next-line no-await-in-loop
+                                await this.tripleStoreService.deleteAssetMetadata(
+                                    paranetRepository,
+                                    blockchain,
+                                    contract,
+                                    tokenId,
+                                );
+                                // eslint-disable-next-line no-await-in-loop
+                                await this.tripleStoreService.deleteAssertion(
+                                    paranetRepository,
+                                    cachedData.public.assertionId,
+                                );
+                                throw error;
+                            }
+                        }
+                    }
+                }
+                if (cachedData.private?.assertion && cachedData.private?.assertionId) {
+                    let attempts = 0;
+                    while (attempts < LOCAL_INSERT_MAX_ATTEMPTS) {
+                        try {
+                            // eslint-disable-next-line no-await-in-loop
+                            await this.tripleStoreService.localStoreAsset(
+                                paranetRepository,
+                                cachedData.private.assertionId,
+                                cachedData.private.assertion,
+                                blockchain,
+                                contract,
+                                tokenId,
+                                keyword,
+                            );
+                            break;
+                        } catch (error) {
+                            attempts += 1;
+                            if (attempts < LOCAL_INSERT_MAX_ATTEMPTS) {
+                                // eslint-disable-next-line no-await-in-loop
+                                await setTimeout(LOCAL_INSERT_RETRY_DELAY);
+                            } else {
+                                // eslint-disable-next-line no-await-in-loop
+                                await this.tripleStoreService.deleteAssetMetadata(
+                                    paranetRepository,
+                                    blockchain,
+                                    contract,
+                                    tokenId,
+                                );
+                                // eslint-disable-next-line no-await-in-loop
+                                await this.tripleStoreService.deleteAssertion(
```

Comment:
If local store fails, doesn't it mean that we haven't inserted it?

Reply:
Local store inserts content and metadata we don't know which failed